test/acceptance/cli-test.js

```diff
@@ -1317,6 +1323,52 @@ describe('ember-template-lint executable', function () {
         expect(result.stderr).toBeFalsy();
       });
     });
+
+    describe('with --max-warnings param', function () {
+      it('should should return exit code 1 if total error is less than max-warnings', async function () {
+        project.setConfig({
+          rules: {
+            'no-bare-strings': true,
+            'no-html-comments': true,
+          },
+        });
+        project.write({
+          app: {
+            templates: {
+              'application.hbs':
+                '<h2>Here too!!</h2><div>Bare strings are bad...</div><!-- bad html comment! -->',
+            },
+          },
+        });
+
+        let result = await run(['.', '--max-warnings=3']); // One less than total
+
+        expect(result.exitCode).toEqual(1);
+        expect(result.stderr).toBeFalsy();
+      });
```

Comment:
Same here, this test suggests at the issue I pointed out in the bin script logic but doesn't actually test it. This test should be about ensuring that _any_ errors cause exitCode === 1 (regardless of `--max-warnings` value).

Can you update to:

```suggestion
      it('should should return exit code 1 if total error is less than max-warnings', async function () {
        project.setConfig({
          rules: {
            'no-bare-strings': 'error',
            'no-html-comments': 'error',
          },
        });
        project.write({
          app: {
            templates: {
              'application.hbs':
                '<h2>Here too!!</h2><div>Bare strings are bad...</div><!-- bad html comment! -->',
            },
          },
        });

        let result = await run(['.', '--max-warnings=10000']);

        expect(result.exitCode).toEqual(1);
        expect(result.stderr).toMatchInlineSnapshot();
      });
```